crates/ruff_python_formatter/src/expression/string/docstring.rs

```diff
@@ -855,6 +1241,58 @@ fn indentation_length(line: &str) -> TextSize {
     TextSize::new(indentation)
 }
 
+/// Trims at most `indent_len` indentation from the beginning of `line`.
+///
+/// This treats indentation in precisely the same way as `indentation_length`.
+/// As such, it is expected that `indent_len` is computed from
+/// `indentation_length`. This is useful when one needs to trim some minimum
+/// level of indentation from a code snippet collected from a docstring before
+/// attempting to reformat it.
+fn indentation_trim(indent_len: TextSize, mut line: &str) -> &str {
+    let mut seen_indent_len = 0u32;
+    for char in line.chars() {
+        if seen_indent_len >= indent_len.to_u32() {
+            return line;
+        }
+        if char == '\t' {
+            // Pad to the next multiple of tab_width
+            seen_indent_len += 8 - (seen_indent_len.rem_euclid(8));
+            line = &line[1..];
+        } else if char.is_whitespace() {
+            seen_indent_len += u32::from(char.text_len());
+            line = &line[char.len_utf8()..];
```

Comment:
Nit: I had to think about whether assigning to `line` here would impact the `line.chars()` call at the top of the loop. It does not, but I would find it easier to understand if we assign the trimmed result to a new variable instead of mutating `line` in place (by adding a `let mut trimmed = line` at the top)

Reply:
Ah yeah you'd get a borrow checker error if it were otherwise. The key is that we're just re-binding a name.

In any case, I've added a new `trimmed` variable.